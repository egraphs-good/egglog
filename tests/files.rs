use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
