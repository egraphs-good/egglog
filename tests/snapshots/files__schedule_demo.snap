---
source: tests/files.rs
expression: serialized
---
nodes:
  Unit-0:
    op: ()
    children: []
    eclass: Unit-0
    cost: 0
  i64-0:
    op: "0"
    children: []
    eclass: i64-0
    cost: 0
  left-0:
    op: left
    children:
      - i64-0
    eclass: Unit-0
    cost: 1
  i64-5871781006564002453:
    op: "1"
    children: []
    eclass: i64-5871781006564002453
    cost: 0
  left-5871781006564002453:
    op: left
    children:
      - i64-5871781006564002453
    eclass: Unit-0
    cost: 1
  i64-11743562013128004906:
    op: "2"
    children: []
    eclass: i64-11743562013128004906
    cost: 0
  left-11743562013128004906:
    op: left
    children:
      - i64-11743562013128004906
    eclass: Unit-0
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  left-17615343019692007359:
    op: left
    children:
      - i64-17615343019692007359
    eclass: Unit-0
    cost: 1
  i64-5040379952546458196:
    op: "4"
    children: []
    eclass: i64-5040379952546458196
    cost: 0
  left-5040379952546458196:
    op: left
    children:
      - i64-5040379952546458196
    eclass: Unit-0
    cost: 1
  i64-10912160959110460649:
    op: "5"
    children: []
    eclass: i64-10912160959110460649
    cost: 0
  left-10912160959110460649:
    op: left
    children:
      - i64-10912160959110460649
    eclass: Unit-0
    cost: 1
  i64-16783941965674463102:
    op: "6"
    children: []
    eclass: i64-16783941965674463102
    cost: 0
  left-16783941965674463102:
    op: left
    children:
      - i64-16783941965674463102
    eclass: Unit-0
    cost: 1
  i64-4208978898528913939:
    op: "7"
    children: []
    eclass: i64-4208978898528913939
    cost: 0
  left-4208978898528913939:
    op: left
    children:
      - i64-4208978898528913939
    eclass: Unit-0
    cost: 1
  i64-10080759905092916392:
    op: "8"
    children: []
    eclass: i64-10080759905092916392
    cost: 0
  left-10080759905092916392:
    op: left
    children:
      - i64-10080759905092916392
    eclass: Unit-0
    cost: 1
  i64-15952540911656918845:
    op: "9"
    children: []
    eclass: i64-15952540911656918845
    cost: 0
  left-15952540911656918845:
    op: left
    children:
      - i64-15952540911656918845
    eclass: Unit-0
    cost: 1
  right-0:
    op: right
    children:
      - i64-0
    eclass: Unit-0
    cost: 1
  right-5871781006564002453:
    op: right
    children:
      - i64-5871781006564002453
    eclass: Unit-0
    cost: 1
  right-11743562013128004906:
    op: right
    children:
      - i64-11743562013128004906
    eclass: Unit-0
    cost: 1
  right-17615343019692007359:
    op: right
    children:
      - i64-17615343019692007359
    eclass: Unit-0
    cost: 1
  right-5040379952546458196:
    op: right
    children:
      - i64-5040379952546458196
    eclass: Unit-0
    cost: 1
  right-10912160959110460649:
    op: right
    children:
      - i64-10912160959110460649
    eclass: Unit-0
    cost: 1
  right-16783941965674463102:
    op: right
    children:
      - i64-16783941965674463102
    eclass: Unit-0
    cost: 1
  right-4208978898528913939:
    op: right
    children:
      - i64-4208978898528913939
    eclass: Unit-0
    cost: 1
  right-10080759905092916392:
    op: right
    children:
      - i64-10080759905092916392
    eclass: Unit-0
    cost: 1
  right-15952540911656918845:
    op: right
    children:
      - i64-15952540911656918845
    eclass: Unit-0
    cost: 1
root_eclasses: []
class_data:
  Unit-0:
    type: Unit
  i64-0:
    type: i64
  i64-5871781006564002453:
    type: i64
  i64-11743562013128004906:
    type: i64
  i64-17615343019692007359:
    type: i64
  i64-5040379952546458196:
    type: i64
  i64-10912160959110460649:
    type: i64
  i64-16783941965674463102:
    type: i64
  i64-4208978898528913939:
    type: i64
  i64-10080759905092916392:
    type: i64
  i64-15952540911656918845:
    type: i64

